browser/src/control/jsdialog/Widget.IconView.ts

```diff
@@ -107,15 +107,23 @@ function _iconViewEntry(
 
 		// Add tabindex attribute for accessibility, enabling keyboard navigation in the icon preview
 		entryContainer.setAttribute('tabindex', '0');
-		JSDialog.OnDemandRenderer(
-			builder,
-			parentData.id,
-			'iconview',
-			entry.row,
-			placeholder,
-			icon,
-			entry.text,
-		);
+		const setupOnDemandRenderer = () => {
+			JSDialog.OnDemandRenderer(
```

Comment:
maybe we could add this inside `JSDialog.OnDemandRenderer` ?
then it will work with all widgets which use that helper

https://github.com/CollaboraOnline/online/blob/8235c62b331bc92bd3b61db390a649ef004634fd/browser/src/control/jsdialog/Util.OnDemandRenderer.ts#L19